### PR TITLE
Lazy init useradd and remove init()

### PR DESCRIPTION
**Pull request by estesp**
This should not have been in init() as it causes these lookups to happen
in all reexecs of the Docker binary. The only time it needs to be
resolved is when a user is added, which is extremely rare.

Docker-DCO-1.1-Signed-off-by: Phil Estes estesp@linux.vnet.ibm.com

zootopia-sloth

### CODE REVIEW

**Comment by estesp** [comment]
Thanks to @cpuguy83 tweeting about `init()` I remembered that I had wanted to do this a long time ago :-]

File: pkg/idtools/usergroupadd_linux.go [hunk 1]
```diff
@@ -95,6 +88,14 @@ func AddNamespaceRangesUser(name string) (int, int, error) {
 
 func addUser(userName string) error {
```
Comment (LK4D4) [inline pkg/idtools/usergroupadd_linux.go]: could you remove empty line pls? :)

Comment (estesp) [inline pkg/idtools/usergroupadd_linux.go]: yes! old people are slow learners ;)

**Comment by LK4D4** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Comment by vieux** [comment]
SGTM

**Comment by thaJeztah** [comment]
> as it causes these lookups to happen in all reexecs of the Docker binary.

@LK4D4 is this something we need for 1.11?

**Comment by LK4D4** [comment]
@thaJeztah nope

**Comment by thaJeztah** [comment]
oh perhaps @tonistiigi as well (who's looking in some optimizations) ^^

**Comment by thaJeztah** [comment]
Alright, thanks! (My last comment just crossed yours)

**Comment by vdemeester** [comment]
Build userns is OK, and windowsTP5 failure unrelated
Merging

**Pull request merged**